pkg/apis/apps/v1alpha1/nodeimage_types.go

```diff
@@ -0,0 +1,193 @@
+	// Specifies the image tag
+	Tag string `json:"tag"`
+
+	// Specifies the create time of this tag
+	// +optional
+	CreatedAt *metav1.Time `json:"createdAt,omitempty"`
+
+	// PullPolicy is an optional field to set parameters of the pulling task. If not specified,
+	// the system will use the default values.
+	// +optional
+	PullPolicy *ImageTagPullPolicy `json:"pullPolicy,omitempty"`
+
+	// List of objects depended by this object. If this image is managed by a controller,
+	// then an entry in this list will point to this controller.
+	// +optional
+	OwnerReferences []v1.ObjectReference `json:"ownerReferences,omitempty"`
+}
+
+// ImageTagPullPolicy defines the policy of the pulling task
+type ImageTagPullPolicy struct {
+	// Specifies the timeout of the pulling task.
+	// Defaults to 600
+	// +optional
+	TimeoutSeconds *int32 `json:"timeoutSeconds,omitempty"`
+
+	// Specifies the number of retries before marking the pulling task failed.
+	// Defaults to 3
+	// +optional
+	BackoffLimit *int32 `json:"backoffLimit,omitempty"`
+
+	// TTLSecondsAfterFinished limits the lifetime of a pulling task that has finished execution (either Complete or Failed).
+	// If this field is set, ttlSecondsAfterFinished after the task finishes, it is eligible to be automatically deleted.
+	// If this field is unset, the task won't be automatically deleted.
+	// If this field is set to zero, the task becomes eligible to be deleted immediately after it finishes.
+	// +optional
+	TTLSecondsAfterFinished *int32 `json:"ttlSecondsAfterFinished,omitempty"`
+
+	// ActiveDeadlineSeconds specifies the duration in seconds relative to the startTime that the task may be active
+	// before the system tries to terminate it; value must be positive integer.
+	// if not specified, the system will never terminate it.
+	// +optional
+	ActiveDeadlineSeconds *int64 `json:"activeDeadlineSeconds,omitempty"`
+}
+
+// NodeImageStatus defines the observed state of NodeImage
+type NodeImageStatus struct {
+	// ObservedGeneration is the most recent generation observed for this Node.
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+
+	// The desired number of pulling tasks, this is typically equal to the number of images in spec.
+	Desired int32 `json:"desired"`
+
+	// The number of pulling tasks which reached phase Succeeded.
+	// +optional
+	Succeeded int32 `json:"succeeded"`
+
+	// The number of pulling tasks  which reached phase Failed.
+	// +optional
+	Failed int32 `json:"failed"`
+
+	// The number of pulling tasks which are not finished.
+	// +optional
+	Pulling int32 `json:"pulling"`
+
+	// all statuses of active image pulling tasks
+	ImageStatuses map[string]ImageStatus `json:"imageStatuses,omitempty"`
+}
+
+// ImageStatus defines the pulling status of an image
+type ImageStatus struct {
+	// Represents statuses of pulling tasks on this node
+	Tags []ImageTagStatus `json:"tags"`
+}
+
+// ImageTagStatus defines the pulling status of an image tag
+type ImageTagStatus struct {
```

Comment:
Cloud we add `imageID` in ImageTagStatus? It is really useful for in-place update.